src/Workspaces/Core/MSBuild.BuildHost/MSBuild/ProjectFile/ProjectFile.cs

```diff
@@ -391,132 +391,75 @@ public void RemoveDocument(string filePath)
-                }
-                else if (IsFrameworkReferenceAssembly(peRef.FilePath))
-                {
-                    // just use short name since this will be resolved by msbuild relative to the known framework reference assemblies.
-                    var fileName = identity != null ? identity.Name : Path.GetFileNameWithoutExtension(peRef.FilePath);
-                    _loadedProject.AddItem(ItemNames.Reference, fileName, metadata);
-                }
-                else // other location -- need hint to find correct assembly
-                {
-                    var relativePath = PathUtilities.GetRelativePath(_loadedProject.DirectoryPath, peRef.FilePath);
-                    var fileName = Path.GetFileNameWithoutExtension(peRef.FilePath);
-                    metadata.Add(MetadataNames.HintPath, relativePath);
-                    _loadedProject.AddItem(ItemNames.Reference, fileName, metadata);
-                }
-            }
-        }
+            var metadata = new Dictionary<string, string>();
+            if (!properties.Aliases.IsEmpty)
+                metadata.Add(MetadataNames.Aliases, string.Join(",", properties.Aliases));
 
-        private static bool IsInGAC(string filePath)
-        {
-            return GlobalAssemblyCacheLocation.RootLocations.Any(static (gloc, filePath) => PathUtilities.IsChildPath(gloc, filePath), filePath);
-        }
-
-        private static string? s_frameworkRoot;
-        private static string FrameworkRoot
-        {
-            get
-            {
-                if (RoslynString.IsNullOrEmpty(s_frameworkRoot))
-                {
-                    var runtimeDir = System.Runtime.InteropServices.RuntimeEnvironment.GetRuntimeDirectory();
-                    s_frameworkRoot = Path.GetDirectoryName(runtimeDir); // back out one directory level to be root path of all framework versions
-                }
-
-                return s_frameworkRoot ?? throw new InvalidOperationException($"Unable to get {nameof(FrameworkRoot)}");
-            }
-        }
+            if (hintPath is not null)
+                metadata.Add(MetadataNames.HintPath, hintPath);
 
-        private static bool IsFrameworkReferenceAssembly(string filePath)
-        {
-            return PathUtilities.IsChildPath(FrameworkRoot, filePath);
+            _loadedProject.AddItem(ItemNames.Reference, metadataReferenceIdentity, metadata);
         }
 
-        public void RemoveMetadataReference(MetadataReference reference, AssemblyIdentity identity)
+        public void RemoveMetadataReference(string shortAssemblyName, string fullAssemblyName, string filePath)
         {
             if (_loadedProject is null)
             {
                 return;
             }
 
-            if (reference is PortableExecutableReference peRef && peRef.FilePath != null)
+            var item = FindReferenceItem(shortAssemblyName, fullAssemblyName, filePath);
+            if (item != null)
```

Comment:
it is a bug if we can't find the item?

Reply:
actually, looks like FindReferenceItem throws if it can't find the item.  so do you need htis check here?

Reply:
Going to defer this along with Todd's comments for cleanup: I believe you are correct, but this code isn't new. It was also throwing prior to this PR and also doing a null check. Will investigate once we have additional tests in place.